### ISSUE TITLE
jax.random.poisson incorrectly labels parameter as rate parameter

### ISSUE THREAD

**Issue by NeilGirdhar** [google/jax#6316]
```python
import jax.numpy as jnp
from jax.random import poisson, PRNGKey, split
from jax import vmap

rng = PRNGKey(123)
rngs = split(rng, 1000)
vp = vmap(poisson, in_axes=(0, None))
print(jnp.mean(vp(rngs, 1.0), axis=0))  # 1.011
print(jnp.mean(vp(rngs, 100.0), axis=0))  # 100.028
print(jnp.mean(vp(rngs, 10000.0), axis=0))  # 9998.968
```
Therefore, the `poisson`'s parameter is a _scale_—not a _rate_ as [described](https://jax.readthedocs.io/en/latest/_autosummary/jax.random.poisson.html#jax.random.poisson).

The easiest solution is to simply change the documentation.

**Comment by NeilGirdhar** [comment]
Minor, I submitted a pull request instead.
